python/python/lance/dataset.py

```diff
@@ -3585,6 +3585,41 @@ def sql(self, sql: str) -> "SqlQueryBuilder":
         """
         return SqlQueryBuilder(self._ds.sql(sql))
 
+    def delta(self) -> "DatasetDeltaBuilder":
```

Comment:
I think python don't really have builders, what about we just have all the builder inputs as function arguments?

Reply:
After thinking about this suggestion. There are two opinions:

* There are some config options that have some check logic in `build` fn:

https://github.com/lance-format/lance/blob/b21872585560cc4538a1f1977d5ac89f799b75c2/rust/lance/src/dataset/delta.rs#L91

Reusing the `DatasetDeltaBuilder` so that we can reuse the check logic?

* The build pattern could keep the scalarity for the future.

WDYT?